### ISSUE TITLE
Translation project completion

### ISSUE THREAD

**Issue by gauravano** [publiclab/plots2#5737]
Here's the list of remaining tasks of Translation project:

- [ ] Translation prompt improvement - https://github.com/publiclab/plots2/issues/5651
- [ ] Updating all the Translation wikis/notes on the publiclab.org
- [ ] Adding documentation for Translation
- [ ]     

cc @jywarren

**Comment by gauravano** [comment]
Updating Translation wikis/notes:
1. https://publiclab.org/wiki/translation - Updated. @jywarren I have replaced the video here as the old video was not available anymore. The current video is more like an introduction to Transifex so if you want I can record a video of translation activity with Public Lab as an example? or, check out https://www.youtube.com/user/TransifexInc/feed and suggest a video which seems relevant. 

2. The second page is https://publiclab.org/notes/liz/10-26-2016/how-to-join-public-lab-s-transifex-project a note written by @ebarry. I have some improvements in mind for that. Can I edit that note? 

Thanks!

**Comment by ebarry** [comment]
hi @gauravano i have made you a co-author of that note, please edit as you see fit! Thanks!

**Comment by gauravano** [comment]
Thanks, @ebarry !!

**Comment by gauravano** [comment]
Hey @publiclab/community-reps, I have did some changes to Language dropdown and also added new languages, check it out :point_down: 

![footer_translate_design](https://user-images.githubusercontent.com/20878070/58649483-d875b000-8329-11e9-85fa-d1e7a6f2bfc2.gif)

What do you think about this design? If you like it, I will update the feature. Thanks!

**Comment by ebarry** [comment]
i like it, and look forward to a future moment when we make the language option more visible :)

**Comment by gauravano** [comment]
I am thinking of an enhancement -- Suppose a user browses a website from China then using their location we can show this language menu to them in the notice or somewhere at top of the website? What do you think @jywarren @ebarry?

Also, I have updated https://publiclab.org/notes/liz/10-26-2016/how-to-join-public-lab-s-transifex-project note with the latest screenshots.  

@ebarry there are some more research notes tagged `translation`. I will go through all of them and so want to ask can I update wherever necessary? you don't have to go through the trouble of making me co-author at all of them as I already have admin privileges so can edit those notes, but just want to ask a would that be fine with you? Thanks!!

**Comment by jywarren** [comment]
Are the dropdown changes in a feature, and do they need to be added to the live site? Cool!

One thing if you haven't already, is to clearly show people how to enable the "see translation prompts everywhere" setting -- which, ok it was a while ago -- i think was driven from a profile tag? We could advertise this and get some language growth going, then consider switching it from an opt-in to an opt-out to see it really grow?

**Comment by gauravano** [comment]
> Are the dropdown changes in a feature, and do they need to be added to the live site? Cool!

yes! Here's the gist - https://gist.github.com/gauravano/1124fff46ae156537e6fb712bff9e980

@jywarren yes, currently profile tag enables the translation prompts. Yes, I am thinking of writing a note. What do you think? For making it easy to enable translation prompts, we can add a button on the profile page or somewhere else?

Thanks!

**Comment by jywarren** [comment]
So, with the right profile tag, should I be seeing the prompts on every
page?

**Comment by jywarren** [comment]
Footer updated! Wow!

**Comment by jywarren** [comment]
sure, yeah! Also, the "people" grid feature on this page could be used to
prompt people to add that profile tag to themselves:

[people:PROFILETAGNAME]

https://publiclab.org/wiki/education#Education+group

**Comment by gauravano** [comment]
> So, with the right profile tag, should I be seeing the prompts on every
page?

Only on pages where I replaced default `t` helper with custom `translation` helper like this https://github.com/publiclab/plots2/blob/master/app/views/dashboard/dashboard.html.erb#L47 . I will document it and some more things today.

> sure, yeah! Also, the "people" grid feature on this page could be used to
prompt people to add that profile tag to themselves:

@jywarren you added such button here - https://publiclab.org/wiki/translation a month ago. Should we add it to more pages?

**Comment by jywarren** [comment]
LOL omg i totally did. So, no problem.

Ah ok re: t vs. translation -- are we going to roll this out across all
templates on a schedule? Thanks, sorry i was out of sync with you and even
myself :-P

**Comment by gauravano** [comment]
We need to change `t` to `raw translation` in views. We can do it wherever we want. Currently, they are on /blog, /dashboard and /subscriptions. I can add them everywhere. What do you think about that?

Thanks!

**Comment by jywarren** [comment]
Yes, let's do the `t` to `translation` conversion now, I'm pretty happy with how this is working, and people can turn it on and off via profile tags anyways! Thanks, this is AWESOME Gaurav!

> I am thinking of an enhancement -- Suppose a user browses a website from China then using their location we can show this language menu to them in the notice or somewhere at top of the website? What do you think @jywarren @ebarry?

I think this sounds good but i think we can detect the languages settings of the user's browser, rather than guessing based on location. That way if someone doesn't speak one of the primary languages of a country their browser settings will still try to show them the site in that language.

**Comment by jywarren** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language maybe?

**Comment by gauravano** [comment]
Sure, headers are always helpful. I will start on this enhancement after updating the notes/wikis related to translation. @jywarren @ebarry so can I go ahead and update the research notes using my admin privileges?

**Comment by jywarren** [comment]
Yes please go ahead! You can add yourself as co-author too!

**Comment by gauravano** [comment]
Thanks, @jywarren for confirming! Also, don't you think what I am currently doing is somewhat similar to @steviepubliclab's note -  https://publiclab.org/notes/stevie/05-24-2019/idea-to-distribute-tasks-for-a-reserach-area-review. But, I think, it'll cover only wikis but if that covers research notes too, then co-authors would be added or the reviewers need to be given admin privileges :thinking:. 

Will update in 2 days and then throw final documentation for pushing and pulling translations and helper methods.

**Comment by jywarren** [comment]
Huh yes it is related! Ok, awesome.

**Comment by jywarren** [comment]
Ok just confirming that any `t(...)` can be replaced with a new, special `translations(...)` method in Ruby, as in this comment by @gauravano?

If so, @publiclab/gci I think this would make a lot of really nice GCI tasks!

**Comment by jywarren** [comment]
Finding instances of the old t() method is a little harder than I thought, as the GitHub search doesn't work, so I think a "grep" based search in Linux would work but it's not as easy to do:

```
grep -r "t(" app/views
```

It should return lines like this one!

https://github.com/publiclab/plots2/blob/d653be356eafa7f6181fb05e2d1e3ac46a2199e5/app/views/notes/show.html.erb#L42

**Comment by nstjean** [comment]
I am having success in Visual Basic Code by searching for `\st\(` - space t (

**Comment by gauravano** [comment]
> Ok just confirming that any t(...) can be replaced with a new, special translations(...) method in Ruby, as in this comment by @gauravano?

Hi @jywarren @nstjean, yeah, `t(..)` can be replaced by `translation(...)`. The code for translation is in `helpers/application_helper.rb`. 

**NOTE:** After replacing, testing that view is necessary as it may not work for complex blocks live div, span, etc. 

And, if corner cases need to be handled, then one can change the helper definition. 

All the instances can be found from RubyMine too or any editor as @nstjean also mentioned an example. I will say, create a list of files here as a checklist and then create issue so as to have no confusion or duplicate issues.

Thanks!
